src/python/pants/engine/legacy/graph.py

```diff
@@ -490,6 +504,30 @@ def owns_any_source(legacy_target):
   return TransitiveHydratedTarget(root, dependencies)
 
 
+@rule
+async def sort_targets(targets: HydratedTargets) -> TopologicallyOrderedTargets:
+  return TopologicallyOrderedTargets(HydratedTargets(topo_sort(tuple(targets))))
+
+
+def topo_sort(targets: Tuple[HydratedTarget, ...]) -> Tuple[HydratedTarget, ...]:
```

Comment:
Nit: this can take an `Iterable[HydratedTarget]` to be more flexible. It doesn't matter, for example, if it was an unordered set. This makes the call site more flexible and ergonomic in some places.

It should definitely still return `Tuple[HydratedTarget, ...]`, though. That's helpful to give more information to call sites on what they'll get back.